app/forms/nsm/steps/disbursement_cost_form.rb

```diff
@@ -4,13 +4,13 @@ class DisbursementCostForm < ::Steps::BaseFormObject
       include DisbursementCosts
       attr_writer :apply_vat
 
-      attribute :miles, :decimal, precision: 10, scale: 3
-      attribute :total_cost_without_vat, :decimal, precision: 10, scale: 2
+      attribute :miles, :fully_validatable_decimal, precision: 10, scale: 3
+      attribute :total_cost_without_vat, :fully_validatable_decimal, precision: 10, scale: 2
       attribute :details, :string
       attribute :prior_authority, :value_object, source: YesNoAnswer
 
-      validates :miles, presence: true, numericality: { greater_than: 0 }, unless: :other_disbursement_type?
-      validates :total_cost_without_vat, presence: true, numericality: { greater_than: 0 },
+      validates :miles, presence: true, is_a_decimal: { greater_than: 0 }, unless: :other_disbursement_type?
```

Comment:
How come we can't do what we do for fully_validatable_integers and gbps, i.e. `validates :miles, presence: true, is_a_number: true, numericality: { greater_than: 0 }`? I'm not sure I understand why we need to add an IsADecimal validator when `is_a_number` and `numericality` together give us what we need.

Reply:
thanks for interrogating this, the problem was due to code I fixed when I was adding tests (weird casting issues in the decimal type) and I can actually get ride of the IsADecimal validator and the other code you mentioned below on the calculation_rows method completely